core/embed/trezorhal/secret.h

```diff
@@ -55,30 +52,15 @@ secbool secret_optiga_set(const uint8_t secret[SECRET_OPTIGA_KEY_LEN]);
 // was made unavailable by calling secret_optiga_hide
 secbool secret_optiga_get(uint8_t dest[SECRET_OPTIGA_KEY_LEN]);
 
-// Backs up the optiga pairing secret from the secret storage to the backup
-// register
-void secret_optiga_backup(void);
-
-// Deletes the optiga pairing secret from the register
-void secret_optiga_hide(void);
-
-// Locks the BHK register. Once locked, the BHK register can't be accessed by
-// the software. BHK is made available to the SAES peripheral
-void secret_bhk_lock(void);
-
-// Verifies that access to the register has been disabled
-secbool secret_bhk_locked(void);
-
 // Regenerates the BHK and writes it to the secret storage
 void secret_bhk_regenerate(void);
 
-// Provision the secret BHK from the secret storage to the BHK register
-// which makes the BHK usable for encryption by the firmware, without having
-// read access to it.
-void secret_bhk_provision(void);
+// Prepares the secret storage for running the firmware
+// Provisions secrets/keys to the firmware, depending on the trust level
+// Disables access to the secret storage until next reset, if possible
+// This function is called by the bootloader before starting the firmware
+void secret_prepare_fw(secbool allow_run_with_secret, secbool trust_all);
 
-// Checks that the optiga pairing secret is present in the secret storage.
-// This functions only works when software has access to the secret storage,
-// i.e. in bootloader. Access to secret storage is restricted by calling
-// secret_hide.
-secbool secret_optiga_present(void);
+// Shows a screen informing the user that installation of custom firmware is
+// currently restricted
+void secret_show_install_restricted_screen(void);
```

Comment:
It appears to me that this function might be better suited for inclusion in bootui.h/bootui.c, where it was originally located. 

It seems more aligned with the functionalities of the bootloader UI than with this module. Additionally, having this function implemented here leads to three identical implementations for f4, u5, and unix - I'm uncertain if they might diverge in the future.

Perhaps `secret_prepare_fw()` could return a `secbool` to indicate whether `secret_show_install_restricted_screen` needs to be called. This adjustment would help reduce the dependency of `secret.c` on `bootui.c`.

Reply:
The dependency on bootui is one thing that i need to avoid, as this needs to compile in firmware and boardloader too.

The implementation on f4 and u5 is actually different, and basically this should never be shown on U5 models, so i can't really have same logic in bootloader code - the F4 needs to display this screen and shutdown, while U5 can just hides the secret - so returning a secbool does not seem like an option.

The f4 and unix implementation is same, but its also symlink. In future, we might want unix implementation to emulate the behavior of u5 or f4 depending on the model its emulating, so these might diverge too.